nodejs/eks/cluster.ts

```diff
@@ -223,14 +225,37 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("instanceRole and instanceRoles are mutually exclusive, and cannot both be set.");
     }
 
-    // If no VPC was specified, use the default VPC.
+    if (args.subnetIds && (args.publicSubnetIds || args.privateSubnetIds)) {
+        throw new Error("subnetIds, and the use of publicSubnetIds and/or privateSubnetIds are mutually exclusive. Choose a single approach.");
+    }
+
+    // Configure default networking architecture.
     let vpcId: pulumi.Input<string> = args.vpcId!;
-    let subnetIds: pulumi.Input<pulumi.Input<string>[]> = args.subnetIds!;
-    if (args.vpcId === undefined) {
+    const subnetIds: pulumi.Input<pulumi.Input<string>[]> = args.subnetIds!;
+    const publicSubnetIds: pulumi.Input<pulumi.Input<string>[]> = args.publicSubnetIds!;
+    const privateSubnetIds: pulumi.Input<pulumi.Input<string>[]> = args.privateSubnetIds!;
```

Comment:
I don't think we need these locals--they're only ever set to the input args. Instead, the various refs to these can use the arguments directly.

Reply:
Done